goldenmaster/modules/testbed2_module/testbed2/generated/core/manyparaminterface.threadsafedecorator.h

```diff
@@ -0,0 +1,105 @@
+
+// Thread safe access
+auto prop1 = testManyParamInterface->getProp1();
+testManyParamInterface->setProp1(0);
+auto prop2 = testManyParamInterface->getProp2();
+testManyParamInterface->setProp2(0);
+auto prop3 = testManyParamInterface->getProp3();
+testManyParamInterface->setProp3(0);
+auto prop4 = testManyParamInterface->getProp4();
+testManyParamInterface->setProp4(0);
+\endcode
+*/
+class TEST_TESTBED2_EXPORT ManyParamInterfaceThreadSafeDecorator : public IManyParamInterface
+{
+public:
+    /** 
+    * ctor
+    * @param impl The ManyParamInterface object to make thread safe.
+    */
+    explicit ManyParamInterfaceThreadSafeDecorator(std::shared_ptr<IManyParamInterface> impl);
+
+    /** Forwards call to ManyParamInterface implementation. */
+    int func1(int param1) override;
+    /** Forwards call to ManyParamInterface implementation. */
+    std::future<int> func1Async(int param1) override;
+
+    /** Forwards call to ManyParamInterface implementation. */
+    int func2(int param1, int param2) override;
+    /** Forwards call to ManyParamInterface implementation. */
+    std::future<int> func2Async(int param1, int param2) override;
+
+    /** Forwards call to ManyParamInterface implementation. */
+    int func3(int param1, int param2, int param3) override;
+    /** Forwards call to ManyParamInterface implementation. */
+    std::future<int> func3Async(int param1, int param2, int param3) override;
+
+    /** Forwards call to ManyParamInterface implementation. */
+    int func4(int param1, int param2, int param3, int param4) override;
+    /** Forwards call to ManyParamInterface implementation. */
+    std::future<int> func4Async(int param1, int param2, int param3, int param4) override;
+
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    void setProp1(int prop1) override;
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    int getProp1() const override;
+
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    void setProp2(int prop2) override;
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    int getProp2() const override;
+
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    void setProp3(int prop3) override;
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    int getProp3() const override;
+
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    void setProp4(int prop4) override;
+    /** Guards and forwards call to ManyParamInterface implementation. */
+    int getProp4() const override;
+
+    /**
+    * Access to a publisher, use it to subscribe for ManyParamInterface changes and signal emission.
+    * @return The publisher for ManyParamInterface.
+    */
+    IManyParamInterfacePublisher& _getPublisher() const override;
```

Comment:
this is thread safe, right?
in a sense it forwards call to impl which is a shared_ptr so as long as the ThreadSafeDecorator exists it exists too
also now with this commit the publisher is thread safe
maybe the comment should also mention that it is thread safe to use it?